### PR TITLE
Remove throws

### LINKED ISSUE #74: Get rid of many Throws
90+ places in code we throw exception. Would be nice to remove them, we can think of alternative.

### PR DESCRIPTION
**Pull request by spalmer25**
fixes #74

### CODE REVIEW

File: src/keys.c [hunk 1]
```diff
@@ -35,126 +36,112 @@ bool read_bip32_path(buffer_t *buf, bip32_path_t *const out) {
...
+    derivation_type_t derivation_type = path_with_curve->derivation_type;
+    unsigned int derivation_mode = derivation_type_to_derivation_mode(derivation_type);
+    signature_type_t signature_type = derivation_type_to_signature_type(derivation_type);
+    cx_curve_t cx_curve = signature_type_to_cx_curve(signature_type);
+
+    public_key->W_len = 65;
+    public_key->curve = cx_curve;
+
+    // generate corresponding public key
+    CX_CHECK(bip32_derive_with_seed_get_pubkey_256(derivation_mode,
```
Comment (ajinkyaraj-23) [inline src/keys.c:94]: use bip32_derive_get_pubkey_256 if we dont have seed.
```suggestion
    CX_CHECK(bip32_derive_get_pubkey_256(derivation_mode,
```

File: src/keys.c [hunk 2]
```diff
@@ -168,76 +155,104 @@ void public_key_hash(uint8_t *const hash_out,
...
+    bip32_path_t const *const bip32_path = &path_with_curve->bip32_path;
+    derivation_type_t derivation_type = path_with_curve->derivation_type;
+    unsigned int derivation_mode = derivation_type_to_derivation_mode(derivation_type);
+    signature_type_t signature_type = derivation_type_to_signature_type(derivation_type);
+    cx_curve_t cx_curve = signature_type_to_cx_curve(signature_type);
+    uint32_t info;
 
+    switch (signature_type) {
+        case SIGNATURE_TYPE_ED25519: {
+            CX_CHECK(bip32_derive_with_seed_eddsa_sign_hash_256(derivation_mode,
```
Comment (ajinkyaraj-23) [inline src/keys.c:224]: Same here , no seed. 
```suggestion
            CX_CHECK(bip32_derive_eddsa_sign_hash_256(derivation_mode,
```

**Review by ajinkyaraj-23 — COMMENTED** [review]
Review in progress..

File: src/keys.c (hunk 1 above)
Comment (spalmer25) [reply src/keys.c:94]: `bip32_derive_get_pubkey_256` do not have the argument `derivation_mode` that is needed for `ED25519` (`HDW_ED25519_SLIP10`).

I can use `bip32_derive_get_pubkey_256` for the 3 others curves and still use `bip32_derive_with_seed_get_pubkey_256` for `ED25519` but I'm not sure if that is clearer.

Comment (ajinkyaraj-23) [reply src/keys.c:94]: I think derive_with_seed is fine.

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM.
 It will be better to create an issue to convert common tasks in pre-condition and post-condition macros just like in wallet app. FUNC_ENTER /FUNC_EXIT where we initialize the error /define end etc.

**Pull request merged**